client/main.go

```diff
@@ -179,16 +173,20 @@ func run(protocol, addr, id string) error {
 
 		elapsed += dt
 		frameChange := 1.0 / animationRate
-		frame := int(elapsed/frameChange) % len(playerFrames)
-		playerSprite = pixel.NewSprite(playerSheet, playerFrames[frame])
+		frame := 0
+		if len(playerSprite.Frames[g.facing][g.action]) > 0 {
+			frame = int(elapsed/frameChange) % len(playerSprite.Frames[g.facing][g.action])
+		}
+		playerSprite.Sprite.Set(playerSprite.Picture, playerSprite.Frames[g.facing][g.action][frame])
+
```

Comment:
i'd like us to make our own `Sprite` or `Animation` class that wraps this functionaltiy (so we can just call `Player.DrawFrame()`

in fact i think we should define an interface
```go
type Animation interface{
    Animate(deltaTime)
}
```

where `Animate()` encapsulates this behavior.

we can worry about that in a later PR, though (when we have more things to animate).

up to you if you want to define and call that interface in this PR